avocado/core/plugin_interfaces.py

```diff
@@ -320,6 +320,18 @@ async def create_task_output_dir(self, runtime_task):
         :type runtime_task: :class:`avocado.core.task.runtime.RuntimeTask`
         """
 
+    @abc.abstractmethod
```

Comment:
I believe I follow the reasoning for this commit, but I'd suggest it wait for an actual implementation to prove its design.  I'm not confident to ack without an actual implementation.

The mention of `ExternalSpawner` had me looking for previous discussions/issues about it, and not finding anything.  I'd recommend you explain it better what an "External" spawner would be, especially its differences from the current ones.

Reply:
Yes, this is preparation for the future development, when it will be needed to fetch the results from the runner environment. So you are proposing to remove this commit for now and use it when this feature will be needed?

Reply:
Yes, simply because I find it hard to judge/review this one commit without an actual implementation that makes use of it.  Do you have an implementation in progress or in mind?  Maybe even sharing that would be OK.  If not yet, then yes, I think it's better to drop this commit for now.